schemas/json/layout/footer.schema.v1.json

```diff
@@ -0,0 +1,82 @@
+{
+  "$id": "https://altinncdn.no/schemas/json/layout/footer.schema.v1.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Altinn footer",
+  "description": "Schema that describes the footer configuration for Altinn applications.",
+  "type": "object",
+  "additionalProperties": true,
+  "properties": {
+    "footer": {
+      "title": "The footer",
+      "description": "Array of components to be presented in the footer.",
+      "type": "array",
+      "items": {"$ref": "#/definitions/footerLayout"}
+    }
+  },
+  "definitions": {
+    "footerLayout": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "title": "Type",
+          "description": "The component type.",
+          "enum": ["Email", "Link", "Phone", "Text"]
+        }
+      },
+      "required": ["type"],
+      "allOf": [
+        { "if": {"properties": {"type": { "const": "Email"}}}, "then": { "$ref": "#/definitions/emailComponent"}},
+        { "if": {"properties": {"type": { "const": "Link"}}}, "then": { "$ref": "#/definitions/linkComponent"}},
+        { "if": {"properties": {"type": { "const": "Phone"}}}, "then": { "$ref": "#/definitions/phoneComponent"}},
+        { "if": {"properties": {"type": { "const": "Text"}}}, "then": { "$ref": "#/definitions/textComponent"}}
+      ]
+    },
+    "icon": {
+      "type": "string",
+      "title": "Icon",
+      "description": "Icon to show next to the component.",
+      "enum": ["information", "email", "phone"]
+    },
+    "title": {
+      "type": "string",
+      "title": "Title",
+      "description": "The text to display."
+    },
+    "emailComponent": {
+      "properties": {
+        "title": { "$ref": "#/definitions/title" },
+        "target": {
+          "type": "string",
+          "title": "Target",
+          "description": "The email to link to."
+        }
+      }
+    },
+    "linkComponent": {
+      "properties": {
+        "title": { "$ref": "#/definitions/title" },
+        "target": {
+          "type": "string",
+          "title": "Target",
+          "description": "The URL to link to."
+        },
+        "icon": { "$ref": "#/definitions/icon" }
+      }
+    },
+    "phoneComponent": {
+      "properties": {
+        "title": { "$ref": "#/definitions/title" },
+        "target": {
+          "type": "string",
+          "title": "Target",
+          "description": "The phone number to link to."
```

Comment:
Can you use text resource keys in here? Same goes for the `title` property. We should probably include that information in the description.

Reply:
Right, should include that in the description (for title at least, have not added text-resource support for target, do you think it should have?)

Reply:
Not sure if it is needed, but if it's easy to implement it might add value. One immediate use-case I can think of is having different phone numbers depending on language.

Reply:
Would be very easy